### PR TITLE
Fixes #36317 - fix ansible-runner cmdline args

### PR DESCRIPTION
**Pull request by Ron-Lavi**
looks like currently the --cmdline args are not passed properly

### CODE REVIEW

File: lib/smart_proxy_ansible/runner/ansible_runner.rb [hunk 1]
```diff
@@ -200,7 +200,7 @@ def start_ansible_runner
         env['FOREMAN_CALLBACK_DISABLE'] = '1' if @rex_command
         env['SMART_PROXY_ANSIBLE_ENVIRONMENT_FILE'] = Proxy::Ansible::Plugin.settings[:ansible_environment_file]
         command = ['ansible-runner', 'run', @root, '-p', 'playbook.yml']
-        command << '--cmdline' << cmdline unless cmdline.nil?
+        command << "--cmdline='#{cmdline}'" unless cmdline.nil?
```
Comment (evgeni) [inline lib/smart_proxy_ansible/runner/ansible_runner.rb:203]: isn't that identical (besides the case where `cmdline` contains a `'` and it now breaks?)

Comment (Ron-Lavi) [reply lib/smart_proxy_ansible/runner/ansible_runner.rb:203]: not entirely sure, for more context it was originated from this POC: https://github.com/theforeman/smart_proxy_ansible/compare/master...Ron-Lavi:smart_proxy_ansible:vault

Comment (adamruzicka) [inline lib/smart_proxy_ansible/runner/ansible_runner.rb:203]: If you provide tags, don't the inner single quotes conflict with the single quotes used on L218?

Comment (evgeni) [reply lib/smart_proxy_ansible/runner/ansible_runner.rb:203]: I see not more context from that diff?

How do the *executed* (mind you, the logger output is "broken" as it has to concatenate an Array) commands differ in the two cases?

Comment (Ron-Lavi) [reply lib/smart_proxy_ansible/runner/ansible_runner.rb:203]: i will try that again and get back to you xD

Comment (evgeni) [reply lib/smart_proxy_ansible/runner/ansible_runner.rb:203]: Thanks!

Would you mind adding tests to that method while already touching it? Otherwise it's very hard to follow when some subtle behaviour change is added.

Comment (nofaralfasi) [reply lib/smart_proxy_ansible/runner/ansible_runner.rb:203]: According to what I checked, it does conflict with other single quotes if we use those. 

@Ron-Lavi I tried to test the code with your changes from here (even though they are irrelevant) https://github.com/theforeman/smart_proxy_ansible/compare/master...Ron-Lavi:smart_proxy_ansible:vault and it works for me without the changes from this PR (with them it failed). 
So I'm good with closing this PR unless I missed something.

**Comment by Ron-Lavi** [comment]
Thanks @nofaralfasi for testing this!
I guess something fishy can happen when you pass quotes into the cmdline option,
and it took me some time in https://github.com/theforeman/smart_proxy_ansible/compare/master...Ron-Lavi:smart_proxy_ansible:vault to make it work, maybe it was the inner quotes in https://github.com/theforeman/smart_proxy_ansible/compare/master...Ron-Lavi:smart_proxy_ansible:vault#diff-65b556454b9717ae33c08487e6ace03409dadb3c66b54f921b1dee1caccfb142R219

closing for now.
